test/jdk/java/awt/print/PrinterJob/PrintNullString.java

```diff
@@ -21,308 +21,160 @@
+
+        TextCanvas c = new TextCanvas();
+        add("Center", c);
+
+        JButton b = new JButton("Print");
+        add("South", b);
+        b.addActionListener(e -> {
+            PrinterJob pj = PrinterJob.getPrinterJob();
+            if (pj.printDialog()) {
+                pj.setPrintable(c);
+                try {
+                    pj.print();
+                } catch (PrinterException ex) {
+                    ex.printStackTrace();
+                    String msg = "PrinterException: " + ex.getMessage();
+                    JOptionPane.showMessageDialog(b, msg, "Error occurred",
+                            JOptionPane.ERROR_MESSAGE);
+                    PassFailJFrame.forceFail(msg);
+                }
+            }
+        });
+        pack();
     }
 
- }// Sysout  class
-
-/**
-  This is part of the standard test machinery.  It provides a place for the
-   test instructions to be displayed, and a place for interactive messages
-   to the user to be displayed.
-  To have the test instructions displayed, see Sysout.
-  To have a message to the user be displayed, see Sysout.
-  Do not call anything in this dialog directly.
-  */
-class TestDialog extends Dialog {
-
-   TextArea instructionsText;
-   TextArea messageText;
-   int maxStringLength = 80;
-
-   //DO NOT call this directly, go through Sysout
-   public TestDialog( Frame frame, String name )
-    {
-      super( frame, name );
-      int scrollBoth = TextArea.SCROLLBARS_BOTH;
-      instructionsText = new TextArea( "", 15, maxStringLength, scrollBoth );
-      add( "North", instructionsText );
-
-      messageText = new TextArea( "", 5, maxStringLength, scrollBoth );
-      add("Center", messageText);
-
-      pack();
+    private static class TextCanvas extends Panel implements Printable {
+        private final String nullStr = null;
+        private final String emptyStr = "";
+        private final AttributedString emptyAttStr = new AttributedString(emptyStr);
+        private final AttributedCharacterIterator nullIterator = null;
+        private final AttributedCharacterIterator emptyIterator = emptyAttStr.getIterator();
+
+        @Override
+        public void paint(Graphics g) {
+            Graphics2D g2d = (Graphics2D) g;
+            paint(g2d);
+        }
+
+        @Override
+        public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
+            if (pgIndex > 0) {
+                return NO_SUCH_PAGE;
+            }
 
-      show();
-    }// TestDialog()
+            Graphics2D g2d = (Graphics2D) g;
+            g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
+            paint(g2d);
 
-   //DO NOT call this directly, go through Sysout
-   public void printInstructions( String[] instructions )
-    {
-      //Clear out any current instructions
-      instructionsText.setText( "" );
+            return PAGE_EXISTS;
+        }
 
-      //Go down array of instruction strings
+        public void paint(Graphics2D g2d) {
```

Comment:
```suggestion
        private void paint(Graphics2D g2d) {
```
To avoid any confusion with public `paint` and `print`, even though it accepts `Graphics2D` as parameter.